packages/fabric8-ui/src/app/space/app-launcher/services/app-launcher-gitprovider.service.ts

```diff
@@ -142,6 +144,21 @@ export class AppLauncherGitproviderService implements GitProviderService {
     );
   }
 
+  getDetectedBuildRuntime(repoUrl: string): Observable<BuildTool> {
+    const url = this.detectorApiUrl + 'detect/build/' + repoUrl;
```

Comment:
Do you need to `encodeURIComponent(repoUrl)`?

Reply:
Yes, It is required. API is expecting only encoded URL.

Reply:
@vikram-raj confirmed to me that the `repoUrl` is already encoded by the caller.